src/storage/v2/property_store.cpp

```diff
@@ -1037,9 +1192,25 @@ PropertyValue PropertyStore::GetProperty(PropertyId property) const {
   return value;
 }
 
+uint64_t PropertyStore::PropertySize(PropertyId property) const {
+  uint64_t size = 0;
+  const uint8_t *data = nullptr;
+  std::tie(size, data) = GetSizeData(buffer_);
+  if (size % 8 != 0) {
+    // We are storing the data in the local buffer.
+    size = sizeof(buffer_) - 1;
+    data = &buffer_[1];
+  }
+  Reader reader(data, size);
+
+  uint64_t property_size = 0;
+  if (FindSpecificPropertySize(&reader, property, property_size) != ExpectedPropertyStatus::EQUAL) return 0;
```

Comment:
How about throwing exception if the property doesn't exist?

Reply:
If you don't want exception then reimplement using std::optional. You don't need to know then what is the status. I would prefer exception with the specific status check however.